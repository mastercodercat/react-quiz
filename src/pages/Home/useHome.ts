import { useHistory } from "react-router-dom";

import { useAppDispatch } from "store/hooks";
import { fetchQuestions } from "store/quiz";

const useHome = () => {
  const dispatch = useAppDispatch();
  const history = useHistory();

  const handleBegin = async () => {
    await dispatch(fetchQuestions());
    history.push("/quiz/1");
  };

  return {
    handleBegin,
  };
};

export default useHome;
