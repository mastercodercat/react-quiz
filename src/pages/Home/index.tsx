import { Button } from "react-bootstrap";

import Layout from "components/Layout";
import useHome from "./useHome";

const Home = () => {
  const { handleBegin } = useHome();

  return (
    <Layout title="Welcome to the Trivia Challenge!">
      <h3 className="text-center">
        You will be presented with 10 True or False questions
      </h3>
      <h3 className="text-center my-5">Can you score 100%?</h3>
      <Button
        variant="primary"
        onClick={handleBegin}
        className="d-block mx-auto px-4 py-2"
      >
        Begin
      </Button>
    </Layout>
  );
};

export default Home;
