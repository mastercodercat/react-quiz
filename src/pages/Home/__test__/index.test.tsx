import {
  screen,
  act,
  fireEvent,
  waitFor,
  RenderResult,
} from "@testing-library/react";

import renderTest, { mockDispatch } from "test/helpers/renderTest";
import Home from "..";

let wrapper: RenderResult;

describe("Home", () => {
  beforeEach(() => {
    jest.useFakeTimers();
    wrapper = renderTest(<Home />, { routes: ["/"] }, "/");
  });

  afterEach(() => {
    act(() => {
      jest.runOnlyPendingTimers();
    });
    jest.useRealTimers();
  });

  it("should test snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should display home", () => {
    expect(
      screen.queryByText("Welcome to the Trivia Challenge!")
    ).toBeInTheDocument();
    expect(
      screen.queryByText(
        "You will be presented with 10 True or False questions"
      )
    ).toBeInTheDocument();
    expect(screen.queryByText("Can you score 100%?")).toBeInTheDocument();
    expect(screen.queryByText("Begin")).toBeInTheDocument();
  });
});
