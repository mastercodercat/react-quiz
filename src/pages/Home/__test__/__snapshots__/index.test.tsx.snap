// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should test snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <div
          class="my-5 mx-auto p-5 bg-secondary bg-opacity-25 col-lg-6 col-md-8 col-sm-12"
        >
          <div
            class="mb-5 row"
          >
            <div
              class="text-center col-sm-12"
            >
              <h1>
                Welcome to the Trivia Challenge!
              </h1>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-sm-12"
            >
              <h3
                class="text-center"
              >
                You will be presented with 10 True or False questions
              </h3>
              <h3
                class="text-center my-5"
              >
                Can you score 100%?
              </h3>
              <button
                class="d-block mx-auto px-4 py-2 btn btn-primary"
                type="button"
              >
                Begin
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <div
        class="my-5 mx-auto p-5 bg-secondary bg-opacity-25 col-lg-6 col-md-8 col-sm-12"
      >
        <div
          class="mb-5 row"
        >
          <div
            class="text-center col-sm-12"
          >
            <h1>
              Welcome to the Trivia Challenge!
            </h1>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-sm-12"
          >
            <h3
              class="text-center"
            >
              You will be presented with 10 True or False questions
            </h3>
            <h3
              class="text-center my-5"
            >
              Can you score 100%?
            </h3>
            <button
              class="d-block mx-auto px-4 py-2 btn btn-primary"
              type="button"
            >
              Begin
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
